src/hal/src/pass.rs

```diff
@@ -85,37 +89,42 @@ pub type AttachmentId = usize;
 /// Reference to an attachment by index and expected image layout.
 pub type AttachmentRef = (AttachmentId, AttachmentLayout);
 
-///
+/// What other subpasses a particular subpass depends on.
 #[derive(Copy, Clone, Debug, Hash, PartialEq)]
 #[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
 pub enum SubpassRef {
-    ///
+    /// The subpass depends on something that was submitted to the
+    /// queue before the render pass began.
     External,
-    ///
+    /// The subpass depends on another subpass with the given index.
```

Comment:
Probably worth noting that index must be smaller than or equal to the index of the current subpass.
Index refers here to the corresponding subpassdescription index in the renderpass creation